boards/PDM/Src/Io/Io_Efuses.c

```diff
@@ -0,0 +1,227 @@
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection);
+
+/**
+ * Exit fail-safe mode and disable the watchdog timer
+ * @return EXIT_CODE_OK if the efuse exited fail-safe mode
+ *         EXIT_CODE_TIMEOUT if the one of the SPI writes timed-out
+ *         EXIT_CODE_UNIMPLEMENTED if the efuse failed to exit fail-safe mode
+ */
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(void);
+
+/**
+ * Write data to a specific Serial Input register on the Aux1/Aux2 Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2WriteRegister(
+    uint8_t  register_address,
+    uint16_t register_value);
+
+/**
+ * Read data from a specific Serial Output register on the Aux1/Aux2 Efuse
+ * @param register_address Serial Output register being read from
+ * @param register_value register contents of Serial Output register
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ReadRegister(
+    uint8_t   register_address,
+    uint16_t *register_value);
+
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection)
+{
+    uint16_t register_value = 0x0000;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, &register_value));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(register_value, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, register_value));
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(void)
+{
+    // Set WDIN bit for next write
+    // 1_1_00000_00000_0000
+    aux1_aux2_efuse.wdin_bit_to_set = true;
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_STATR_0_ADDR, 0x0000));
+
+    // Disable the watchdog timer
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG));
+
+    // Check if the the efuse is still in fail-safe mode
+    if (HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return EXIT_CODE_UNIMPLEMENTED;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2WriteRegister(
+    uint8_t  register_address,
+    uint16_t register_value)
+{
+    return Io_Efuse_WriteRegister(
```

Comment:
When we call `Io_Efuse_Create` we should also pass the CSB port and pin as inputs and store it in the Efuse_Context struct. Then we can move `Io_Efuse_WriteRegister` to the `Io_Efuse.c` file.

Reply:
Same applies for Io_Efuse_ReadRegister function